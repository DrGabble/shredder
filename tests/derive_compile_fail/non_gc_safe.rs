#[macro_use]
extern crate shredder;

struct NotGcSafe {}

#[derive(Scan)]
struct Test0 {
    #[shredder(skip)]
    field: NotGcSafe
}

fn main() {}
