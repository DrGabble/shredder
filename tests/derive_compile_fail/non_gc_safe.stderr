error[E0277]: the trait bound `NotGcSafe: shredder::GcSafe` is not satisfied
 --> $DIR/non_gc_safe.rs:6:10
  |
6 | #[derive(Scan)]
  |          ^^^^
  |          |
  |          expected an implementor of trait `shredder::GcSafe`
  |          help: consider borrowing here: `&Scan`
  |
  = note: required because of the requirements on the impl of `shredder::GcSafe` for `NotGcSafe`
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
